Lib/test/test_typing.py

```diff
@@ -569,6 +569,15 @@ def test_equal(self):
         self.assertEqual(Literal[1, 2], Literal[2, 1])
         self.assertEqual(Literal[1, 2, 3], Literal[1, 2, 3, 3])
 
+    def test_hash(self):
+        self.assertNotEqual(hash(Literal[0]), hash(Literal[False]))
+        self.assertNotEqual(hash(Literal[True]), hash(Literal[1]))
+        self.assertNotEqual(hash(Literal[1]), hash(Literal[2]))
+        self.assertNotEqual(hash(Literal[1, True]), hash(Literal[1]))
```

Comment:
You're on thin ice here. While we require `a == b` to imply `hash(a) == hash(b)`, we do *not* expect `a != b` to imply `hash(a) != hash(b)`. (It requires some understanding of hash tables to understand this. The basic idea is that there may be hash collisions.)

Reply:
My bad, nice catch.

> While we require a == b to imply hash(a) == hash(b), we do not expect a != b to imply hash(a) != hash(b).

I remembered this rule from computer science classes)